ipsets/ipsets.go

```diff
@@ -792,6 +820,27 @@ func (s *IPSets) ApplyDeletions() {
 	s.gaugeNumIpsets.Set(float64(len(s.ipSetIDToIPSet)))
 }
 
+// tryTempIPSetDeletions tries to delete any temporary IP sets found by the last resync.
+func (s *IPSets) tryTempIPSetDeletions() {
+	s.pendingTempIPSetDeletions.Iter(func(item interface{}) error {
+		setName := item.(string)
+		logCxt := s.logCxt.WithField("setName", setName)
+		if s.existingIPSetNames.Contains(setName) {
+			logCxt.Info("Deleting IP set.")
+			if err := s.deleteIPSet(setName); err != nil {
+				// Log and carry on; we'll try again in ApplyDeletions().
+				logCxt.WithError(err).Warning("Failed to delete temporary IP set. Will retry...")
+			} else {
+				// Success! Remove from the main pending deletions set too.
+				logCxt.WithField("setName", setName).Info("Successfully removed left-over temporary IP set.")
+				s.pendingIPSetDeletions.Discard(item)
```

Comment:
1. Should this be `s.pendingTempIPSetDeletions.Discard(item)` ?

2. What is the difference between that and `return set.RemoveItem` ?  Would it be better to use just one of those options, both times?

Reply:
No difference, concern is removing from a set that we're iterating over.  As it happens, it is safe to remove from a go map while iterating over it but that's assuming a particular implementation of set.